src/plugins/workspace/public/plugin.ts

```diff
@@ -74,11 +77,36 @@ export class WorkspacePlugin implements Plugin<{}, {}> {
     });
   }
 
-  public async setup(core: CoreSetup, { savedObjectsManagement }: WorkspacePluginSetupDeps) {
+  /**
+   * If workspace is enabled and user has entered workspace, hide advance settings and dataSource menu and disable
+   */
+  private disableManagementApps(core: CoreSetup, management: ManagementSetup) {
+    const currentWorkspaceId$ = core.workspaces.currentWorkspaceId$;
+    this.managementCurrentWorkspaceIdSubscription?.unsubscribe();
+
+    this.managementCurrentWorkspaceIdSubscription = currentWorkspaceId$.subscribe(
+      (currentWorkspaceId) => {
+        if (currentWorkspaceId) {
+          const managementSectionApps = management.sections.section.opensearchDashboards.getAppsEnabled();
+          const disabledApps = managementSectionApps.filter(
+            (app) => app.id === 'settings' || app.id === 'dataSources'
```

Comment:
Nit: If there is any consts that we can refer to? if not just ignore me.

Reply:
No. These plugins all use hard code to register. We'd better not define conststants and import from their plugins.